victim-loss-identifier/src/utils/utils.ts

```diff
@@ -0,0 +1,189 @@
+import DataFetcher from "src/fetcher";
+import {
+  FP_MAX_VICTIMS_THRESHOLD,
+  FP_MIN_VICTIMS_THRESHOLD,
+  FP_PROFIT_THRESHOLD,
+  FP_SELLER_TO_BUYER_TXS_THRESHOLD,
+} from "../constants";
+import { ScammerInfo, VictimInfo, FpTransaction, EtherscanApisInterface } from "../types";
+
+export function getChainBlockTime(chainId: number): number {
+  switch (chainId) {
+    case 10: // Optimism
+      return 2;
+    case 56: // BNB Smart Chain
+      return 3;
+    case 137: // Polygon
+      return 2;
+    case 250: // Fantom
+      return 2;
+    case 42161: // Arbitrum
+      return 1;
+    case 43114: // Avalance
+      return 2;
+    default: // Ethereum
+      return 12;
+  }
+}
+
+export function getBlocksInTimePeriodForChainId(timePeriodInSecs: number, chainId: number): number {
+  const chainBlockTime = getChainBlockTime(chainId);
+
+  return Math.floor(timePeriodInSecs / chainBlockTime);
+}
+
+export function cleanObject(object: { [key: string]: ScammerInfo } | { [key: string]: VictimInfo }) {
+  // Sort and delete 1/4 of the keys
+  const sortedKeys = Object.keys(object).sort((a, b) => {
+    const latestTimestampA = object[a].mostRecentActivityByBlockNumber;
+    const latestTimestampB = object[b].mostRecentActivityByBlockNumber;
+    return latestTimestampA - latestTimestampB;
+  });
+
+  // Ensure that at least one key will be deleted in the theoretical edge case where Math.floor(sortedKeys.length / 4) equals 0
+  const indexToDelete = Math.max(1, Math.floor(sortedKeys.length / 4));
+
+  for (let i = 0; i < indexToDelete; i++) {
+    delete object[sortedKeys[i]];
+  }
+}
+
+export function extractFalsePositiveDataAndUpdateState(
+  scammerAddress: string,
+  scammersCurrentlyMonitored: { [key: string]: ScammerInfo },
+  victims: { [key: string]: VictimInfo }
+) {
+  const fpData: FpTransaction[] = [];
+  const fpVictims: string[] = [];
+  const fpScammer = scammersCurrentlyMonitored[scammerAddress];
+
+  for (const victimAddress of Object.keys(fpScammer.victims!)) {
+    const victimInfo = fpScammer.victims![victimAddress];
+    const scammedBy = victimInfo.scammedBy;
+
+    const hasBeenAlerted = scammedBy[scammerAddress].hasBeenAlerted;
+    if (!hasBeenAlerted) continue;
```

Comment:
```suggestion
    const hasBeenAlerted = scammedBy[scammerAddress].hasBeenAlerted;
    // Looking for victim's scammers that have had alerts
    // emitted to create an FP alert to "undo" the first one.
    // If they have no existing alert to "undo", skip
    if (!hasBeenAlerted) continue;
```
Suggestion to add a comment to explain this _peculiar_ `if` check.